src/swift/Wrapper.swift

```diff
@@ -0,0 +1,297 @@
+	}
+
+	internal init(__type: UInt, path: UnsafePointer<Int8>, oflag: Int32,
+				  mode: mode_t, queue: DispatchQueue, handler: (error: Int32) -> Void) {
+		__wrapped = dispatch_io_create_with_path(__type, path, oflag, mode, queue.__wrapped, handler)
+	}
+
+	internal init(__type: UInt, io: DispatchIO,
+				  queue: DispatchQueue, handler: (error: Int32) -> Void) {
+		__wrapped = dispatch_io_create_with_io(__type, io.__wrapped, queue.__wrapped, handler)
+	}
+
+	internal init(queue:dispatch_queue_t) {
+		__wrapped = queue
+	}
+
+	public func barrier(execute: () -> ()) {
+		dispatch_io_barrier(self.__wrapped, execute)
+	}
+
+	public var fileDescriptor: Int32 {
+		return dispatch_io_get_descriptor(__wrapped)
+	}
+
+	public func setLimit(highWater: Int) {
+		dispatch_io_set_high_water(__wrapped, highWater)
+	}
+
+	public func setLimit(lowWater: Int) {
+		dispatch_io_set_low_water(__wrapped, lowWater)
+	}
+}
+
+public class DispatchQueue : DispatchObject {
+	internal let __wrapped:dispatch_queue_t;
+
+	internal override func wrapped() -> dispatch_object_t {
+		return unsafeBitCast(__wrapped, to: dispatch_object_t.self)
+	}
+
+	internal init(__label: String, attr: dispatch_queue_attr_t?) {
+		__wrapped = dispatch_queue_create(__label, attr)
+	}
+
+	internal init(__label: String, attr:  dispatch_queue_attr_t?, queue: DispatchQueue?) {
+		__wrapped = dispatch_queue_create_with_target(__label, attr, queue?.__wrapped)
+	}
+
+	internal init(queue:dispatch_queue_t) {
+		__wrapped = queue
+	}
+
+	public func sync(execute workItem: @noescape ()->()) {
+		dispatch_sync(self.__wrapped, workItem)
+	}
+}
+
+public class DispatchSource : DispatchObject,
+	DispatchSourceType,	DispatchSourceRead,
+	DispatchSourceSignal, DispatchSourceTimer,
+	DispatchSourceUserDataAdd, DispatchSourceUserDataOr,
+	DispatchSourceWrite {
+	internal let __wrapped:dispatch_source_t
+
+	internal override func wrapped() -> dispatch_object_t {
+		return unsafeBitCast(__wrapped, to: dispatch_object_t.self)
+	}
+
+	internal init(source:dispatch_source_t) {
+		__wrapped = source
+	}
+}
+
+#if HAVE_MACH
+extension DispatchSource : DispatchSourceMachSend,
+	DispatchSourceMachReceive, DispatchSourceMemoryPressure {
+}
+#endif
+
+#if !os(Linux)
+extension DispatchSource : DispatchSourceProcess,
+	DispatchSourceFileSystemObject {
+}
+#endif
+
+public typealias DispatchSourceHandler = @convention(block) () -> Void
+
+public protocol DispatchSourceType {
+	func setEventHandler(qos: DispatchQoS, flags: DispatchWorkItemFlags, handler: DispatchSourceHandler?)
+
+	func setEventHandler(handler: DispatchWorkItem)
+
+	func setCancelHandler(qos: DispatchQoS, flags: DispatchWorkItemFlags, handler: DispatchSourceHandler?)
+
+	func setCancelHandler(handler: DispatchWorkItem)
+
+	func setRegistrationHandler(qos: DispatchQoS, flags: DispatchWorkItemFlags, handler: DispatchSourceHandler?)
+
+	func setRegistrationHandler(handler: DispatchWorkItem)
+
```

Comment:
I changed the eventHandler functions of the DispatchSourceType protocol to match the functions defined  in the extension in Source.swift.  This enables the code to compile successfully (ie without triggering an assertion failure in swiftc).  Is this the right change from an API perspective?